spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/core/convert/StructAccessor.java

```diff
@@ -145,14 +147,43 @@ public Object getSingleValue(int colIndex) {
 
   List getListValue(String colName) {
     if (this.struct.getColumnType(colName).getCode() != Code.ARRAY) {
-      throw new SpannerDataException("Column is not an ARRAY type: " + colName);
+      throw new SpannerDataException(EXCEPTION_NARRATIVE_COL_NOT_ARRAY + colName);
     }
     Type.Code innerTypeCode = this.struct.getColumnType(colName).getArrayElementType().getCode();
     Class clazz = SpannerTypeMapper.getSimpleJavaClassFor(innerTypeCode);
     BiFunction<Struct, String, List> readMethod = readIterableMapping.get(clazz);
     return readMethod.apply(this.struct, colName);
   }
 
+  <T> List<T> getListJsonValue(String colName, Class<T> colType) {
+    if (this.struct.getColumnType(colName).getCode() != Code.ARRAY) {
+      throw new SpannerDataException(EXCEPTION_NARRATIVE_COL_NOT_ARRAY + colName);
+    }
+    List<String> jsonStringList = this.struct.getJsonList(colName);
+    List<T> result = new ArrayList<>();
+    jsonStringList.forEach(item ->
+        result.add(gson.fromJson(item, colType)));
+    return result;
+  }
+
+  public  <T> Object getJsonValue(int colIndex, Class<T> colType) {
+    if (this.struct.getColumnType(colIndex).getCode() != Code.ARRAY) {
+      return getSingleJsonValue(colIndex, colType);
+    }
+    return getListJsonValue(colIndex, colType);
+  }
+
+  private  <T> List<T> getListJsonValue(int colIndex, Class<T> colType) {
+    if (this.struct.getColumnType(colIndex).getCode() != Code.ARRAY) {
+      throw new SpannerDataException(EXCEPTION_NARRATIVE_COL_NOT_ARRAY + colIndex);
+    }
+    List<String> jsonStringList = this.struct.getJsonList(colIndex);
+    List<T> result = new ArrayList<>();
+    jsonStringList.forEach(item ->
+        result.add(gson.fromJson(item, colType)));
+    return result;
+  }
```

Comment:
You might be able to reuse the same code for the two `getListJsonValue` definitions by passing `struct -> struct.getJsonList(colIndex);` and `struct->struct.getJsonList(colName)` as lambdas.